### PR TITLE
Feature/initial survey trigger

### PR DESCRIPTION
**Pull request by CJSpohn**
## Is this a fix or a feature?
- Feature

## What is the change/fix?
- Subscribing to a crate takes you to an initial form page
- Clicking the finish button takes you to the subscriptions page
- The survey still needs to be built out but the core concept is there

## Where should the reviewer start?
- modules/user/StyleSurvey.js

## How should this be tested?
- Ensure that the subscribe -> subscriptions page is successfully interrupted by the style survey

### CODE REVIEW

File: code/web/src/modules/crate/Item.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ class Item extends PureComponent {
   }
 
   onClickSubscribe = (crateId) => {
+    /*
```
Comment (kristenmb) [inline code/web/src/modules/crate/Item.js:32]: This is where we will handle the logic of whether or not the user has completed the style survey in the past. This is what will either lead them to the survey view or to subscriptions.

File: code/web/src/modules/crate/Item.js [hunk 2]
```diff
@@ -42,7 +50,9 @@ class Item extends PureComponent {
         } else {
           this.props.messageShow('Subscribed successfully.')
 
-          this.props.history.push(userRoutes.subscriptions.path)
+          // this.props.history.push(userRoutes.subscriptions.path)
```
Comment (kristenmb) [inline code/web/src/modules/crate/Item.js:53]: This was the old route, we commented it out for now to ensure the new path would work correctly to take the user to the survey.

File: code/web/src/modules/user/StyleSurvey.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+class StyleSurvey extends PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoading: false,
+      result: '',
+      //dummydata
+      choices: [1, 2, 3, 4]
```
Comment (kristenmb) [inline code/web/src/modules/user/StyleSurvey.js:29]: Stand in data for the images we will display for each survey page

File: code/web/src/modules/crate/Item.js [hunk 4]
```diff
@@ -42,7 +50,9 @@ class Item extends PureComponent {
...
+          this.props.history.push(userRoutes.styleSurvey.path)
```
Comment (conconartist) [inline code/web/src/modules/crate/Item.js:54]: This is where we can intercept the path from the original subscriptions page to the style survey.

File: code/web/src/modules/user/StyleSurvey.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+                      <div style={{ padding: '1em 1.2em' }}>
+                        <H4 font="secondary" style={{ color: black }}>{choice}</H4>
+
+                        <p style={{ color: grey2, marginTop: '1em' }}>Style Name</p>
+
+                        <p style={{ textAlign: 'center', marginTop: '1.5em', marginBottom: '1em' }}>
+                          <Button
+                            theme="primary"
+                            // onClick={this.onClickSubscribe.bind(this, id)}
```
Comment (kristenmb) [inline code/web/src/modules/user/StyleSurvey.js:70]: We will implement some logic here to allow the user to select their preferred style on each page of the survey

File: code/web/src/modules/user/StyleSurvey.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      <>
+      {/* Top title bar */}
+      <Grid style={{ backgroundColor: grey }}>
+        <GridCell style={{ padding: '2em', textAlign: 'center' }}>
+          <H3 font="secondary">Style Survey!</H3>
+
+          <p style={{ marginTop: '1em', color: grey2 }}>Let's find your style profile!</p>
+        </GridCell>
+      </Grid>
+
```
Comment (conconartist) [inline code/web/src/modules/user/StyleSurvey.js:48]: Copying the styling components seems to work well here! Excited to build out the rest of the functionality.

File: code/web/src/modules/user/StyleSurvey.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                ))
+          }
+        </GridCell>
+      </Grid>
+
+        <Button
+          theme="primary"
+          onClick={this.onClickSubmit.bind(this)}
+          type="button"
+        >Submit</Button>
```
Comment (conconartist) [inline code/web/src/modules/user/StyleSurvey.js:89]: On submit, it should take us to the next survey question and log this response.

**Review by conconartist — APPROVED** [review]

**Pull request merged**